ledger/src/shred.rs

```diff
@@ -34,15 +36,19 @@ pub const SIZE_OF_SIGNATURE: usize = 64;
 pub const SIZE_OF_SHRED_TYPE: usize = 1;
 pub const SIZE_OF_SHRED_SLOT: usize = 8;
 pub const SIZE_OF_SHRED_INDEX: usize = 4;
+pub const SIZE_OF_NONCE: usize = 4;
 pub const SIZE_OF_DATA_SHRED_IGNORED_TAIL: usize =
     SIZE_OF_COMMON_SHRED_HEADER + SIZE_OF_CODING_SHRED_HEADER;
 pub const SIZE_OF_DATA_SHRED_PAYLOAD: usize = PACKET_DATA_SIZE
     - SIZE_OF_COMMON_SHRED_HEADER
     - SIZE_OF_DATA_SHRED_HEADER
     - SIZE_OF_DATA_SHRED_IGNORED_TAIL;
+pub const SIZE_OF_NONCE_DATA_SHRED_PAYLOAD: usize = SIZE_OF_DATA_SHRED_PAYLOAD - SIZE_OF_NONCE;
 
 pub const OFFSET_OF_SHRED_SLOT: usize = SIZE_OF_SIGNATURE + SIZE_OF_SHRED_TYPE;
 pub const OFFSET_OF_SHRED_INDEX: usize = OFFSET_OF_SHRED_SLOT + SIZE_OF_SHRED_SLOT;
+pub const NONCE_SHRED_PAYLOAD_SIZE: usize = PACKET_DATA_SIZE - SIZE_OF_NONCE;
+pub const UNLOCK_NONCE_SLOT: Slot = 13_800_000;
```

Comment:
@mvines Set the target for about 500,000 slots past current Tds, can definitely adjust based on whenever the next planned upgrade is!

Reply:
Let's align with this rolling update (epoch 47) to keep it simple to track: https://github.com/solana-labs/solana/pull/10028#discussion_r425205226

Reply:
But this is actually tricky:
* devnet -- v1.1, current slot: 4384485
* testnet -- v1.1, current slot: 13337978
* mainnet-beta -- v1.0, current slot: 10986442

Plumbing operating mode into here looks to be quite painful.  Here's an alternative:
1. Ship v1.1.x that enables UNLOCK_NONCE_SLOT at testnet epoch 47.  Everybody on testnet upgrades
2. Ship v1.0.z that enables UNLOCK_NONCE_SLOT on mainnet-beta epoch X.  Everybody on mainnet-beta upgrades.  This must occur before mainnet-beta upgrades to v1.1.  
3. After epoch 47, ship v.1.1.y removes UNLOCK_NONCE_SLOT.  It's now always enabled.  This then allows us to migrate devnet, without affecting testnet

UNLOCK_NONCE_SLOT never lands on the master branch.  Nonces are always enabled on the master branch from day 1.